keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1210 @@
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+While updating the e2e load balancer tests after the final [removal of
+in-tree cloud providers], we have run into three problems:
+
+  1. The tests have hard-coded timeouts (that sometimes differ per
+     cloud provider) for deciding how long to wait for the cloud
+     provider to update the service. It would make much more sense for
+     the cloud provider to just provide information about its status
+     on the Service object, so the tests could just monitor that.
+
+  2. The tests recognize that not all cloud providers can implement
+     all load balancer features, but in the past this was handled by
+     hard-coding the information into the individual tests. (e.g.,
+     `e2eskipper.SkipUnlessProviderIs("gce", "gke", "aws")`) These
+     skip rules no longer work in the providerless tree, and this
+     approach doesn't scale anyway. OTOH, we don't want to have to
+     provide a separate `Feature:` tag for each load balancer
+     subfeature, or have each cloud provider have to maintain their
+     own set of `-ginkgo.skip` rules. It would be better if the e2e
+     tests themselves could just figure out, somehow, whether they
+     were running under a cloud provider that intends to implement the
+     feature they are testing, or a cloud provider that doesn't.
+
+  3. In some cases, because the existing tests were only run on
+     certain clouds, it is not clear what the expected semantics are
+     on other clouds. For example, since `IPMode: Proxy` load
```

Comment:
@thockin this was an interesting finding during the implementation of this mode in cloud-provider-kind, we need to try to flesh out more details before going to GA

Reply:
The problem isn't with the `IPMode` KEP; the distinction between "proxy mode" and "VIP mode" already existed before that; it's just that before `IPMode` made it explicit, it was controlled implicitly by whether the LB set `Hostname` or `IP`.

(Which is to say, even if we dropped `IPMode`, the problem would still exist. For example, with the default cloud-provider-aws load balancers.)